src/lib/loaders/api/loader_without_cache_factory.ts

```diff
@@ -1,42 +1,87 @@
 import DataLoader from "dataloader"
 import { loaderInterface } from "./loader_interface"
 import { LoaderFactory } from "../index"
-import { DataLoaderKey } from "./index"
+import { API, DataLoaderKey, APIOptions } from "./index"
+import { warn } from "lib/loggers"
+import gravity from "lib/apis/gravity"
+import extensionsLogger, { formatBytes } from "lib/loaders/api/extensionsLogger"
+import timer from "lib/timer"
 
 /**
  * This factory provides a short-cut system for our data loader system, it provides
  * an uncached result that is close to the equivalent to calling the fetch request directly.
  *
  * @param {(string, any) => Promise<any>} api a function that performs an API request
- * @param {string} _apiName the name of the API service
+ * @param {string} apiName the name of the API service
  */
 
-export const uncachedLoaderFactory = (
-  api: (route: string, params) => Promise<any>,
-  apiName: string
+export const unauthenticatedUncachedApiLoaderFactory = <T = any>(
+  api: API,
+  apiName: string,
+  apiOptions: APIOptions = {}
 ) => {
-  const apiLoaderFactory = (path, options: any | null) => {
-    // If you use gravity as the api here, then options will get interpreted as
-    // an accessToken, so you have to explicitly pass null
-    const loader = new DataLoader<DataLoaderKey, any>(
+  const apiLoaderFactory = (
+    path: string,
+    globalParams = {},
+    pathAPIOptions = {}
+  ) => {
+    const loader = new DataLoader<DataLoaderKey, T | { body: T; headers: any }>(
       (keys) =>
-        Promise.all(
-          keys.map(({ key, apiOptions }) => {
-            if (apiOptions) {
-              throw new Error("A uncachedLoader does not accept API options.")
+        Promise.all<any>(
+          keys.map(({ key, apiOptions: invocationAPIOptions }) => {
+            const options = {
+              ...apiOptions,
+              ...pathAPIOptions,
+              ...invocationAPIOptions,
             }
 
-            const reduceData = ({ body, headers }) =>
-              options && options.headers ? { body, headers } : body
+            const clock = timer(key)
+            clock.start()
 
-            return Promise.resolve(
-              api(key, apiName === "gravity" ? null : options).then(reduceData)
```

Comment:
This line looks to be doing something custom when this factory is used for `gravity` vs. for other services. A couple of lines down tho the rewritten version of the line has this conditional going away. I don't recall the issue (and worry specs aren't catching it and it might be a runtime thing).